### PR TITLE
v12 compatibility fixes

### PR DESCRIPTION
**Pull request by arcos**
I just started plugging away at this so OSE is usable in v12. I am not sure what the plan is, but hopefully this helps. Fixes dice rolls and makes the character sheets update again.

### CODE REVIEW

File: src/module/actor/entity.js [hunk 1]
```diff
@@ -14,7 +14,8 @@ const removeFalsyElements = (arr) =>
 
 export default class OseActor extends Actor {
   prepareDerivedData() {
-    if (game.version.startsWith("10")) this.system.prepareDerivedData?.();
+    //if (game.version.startsWith("10")) this.system.prepareDerivedData?.();
+    this.system.prepareDerivedData?.();
```
Comment (wyrmisis) [inline src/module/actor/entity.js:18]: Did this get fixed in v10's patch cycle? v10 doesn't automatically call the data model's `prepareDerivedData`, but v11 (and later, I imagine) does.

File: src/module/settings.ts [hunk 2]
```diff
@@ -67,6 +67,15 @@ const registerSettings = () => {
...
   });
 
+  game.settings.register(game.system.id, "encumbranceItemStrengthMod", {
+    name: game.i18n.localize("OSE.Setting.EncumbranceItemStrengthMod"),
+    hint: game.i18n.localize("OSE.Setting.EncumbranceItemStrengthModHint"),
+    default: false,
+    scope: "world",
+    type: Boolean,
+    config: true,
+  });
```
Comment (wyrmisis) [inline src/module/settings.ts:77]: Mind doing this as a separate PR?

File: src/ose.js [hunk 3]
```diff
@@ -32,6 +32,10 @@ import * as treasure from "./module/helpers-treasure";
 
 import "./e2e";
 
+if(typeof Math.clamp !== "function") {
+  Math.clamp = Math.clamped;
+}
```
Comment (wyrmisis) [inline src/ose.js:37]: We might as well start a `polyfills.ts` for this kind of stuff.

File: src/ose.js [hunk 4]
```diff
@@ -75,11 +79,11 @@ Hooks.once("init", async () => {
   CONFIG.Actor.documentClass = OseActor;
   CONFIG.Item.documentClass = OseItem;
 
-  CONFIG.Actor.systemDataModels = {
```
Comment (wyrmisis) [inline src/ose.js:78]: Is this going to break things for people on v10 or v11? Same with the next set of changes.

File: src/module/actor/entity.js (hunk 1 above)
Comment (arcos) [reply src/module/actor/entity.js:18]: You may know more than me, but under v12 the `prepareDerivedData` method was not being called (the character sheet form didn't update). It appears to work without issue in v11. It's not clear to me what the change is between v11 and v12 though.

File: src/ose.js (hunk 4 above)
Comment (arcos) [reply src/ose.js]: Reverted, I didn't know that we were keeping v10 support

File: src/module/actor/entity.js (hunk 1 above)
Comment (apewall) [reply src/module/actor/entity.js:18]: It appears that in v12 `prepareDerivedData` is only called automatically if the new TypeDataModel classes are used.
https://github.com/foundryvtt/foundryvtt/issues/10411

Updating the classes to use TypeDataModel should resolve this and make calling prepareDerivedData unnecessary.

https://github.com/vttred/ose/blob/339b156cefd6cc17262a89f95b9d43b2024dd01c/src/ose.js#L78C1-L89C5

Comment (arcos) [reply src/module/actor/entity.js:18]: @apewall Thank you! Will this end support for v10 then? Is that ok?

Comment (apewall) [reply src/module/actor/entity.js:18]: I believe intention is to only support the latest stable (v12) and maintenance (v11) releases of foundryvtt. 

So it should be okay to drop v10 specific code like this.

File: system.json [hunk 5]
```diff
@@ -5,11 +5,11 @@
...
   "version": "AUTOMATICALLY REPLACED BY GITHUB WORKFLOW ACTION",
   "minimumCoreVersion": "10",
-  "compatibleCoreVersion": "11",
+  "compatibleCoreVersion": "12",
   "compatibility": {
     "minimum": "10",
     "verified": "11",
-    "maximum": "11"
+    "maximum": "12"
   },
```
Comment (anthonyronda) [inline system.json:13]: this will probably be changed in a later chore PR

**Review by anthonyronda — APPROVED** [review]
LGTM

**Comment by anthonyronda** [comment]
@all-contributors please add arcos for code

**Pull request merged**